detections/endpoint/multiple_disabled_users_failing_to_authenticate_from_host_using_kerberos.yml

```diff
@@ -0,0 +1,59 @@
+name: Multiple Disabled Users Failing To Authenticate From Host Using Kerberos
+id: 98f22d82-9d62-11eb-9fcf-acde48001122
+version: 1
+date: '2021-04-14'
+author: Mauricio Velazco, Splunk
+type: batch
+datamodel:
+- Endpoint
+description: 'The following analytic identifies one source endpoint failing to authenticate
+  with multiple disabled domain users using the Kerberos protocol. This behavior could
+  represent an adversary performing a Password Spraying attack against an Active Directory
+  environment using Kerberos to obtain initial access or elevate privileges. As attackers
+  progress in a breach, mistakes will be made. I certain scenarios, adversaries may
+  execute a password spraying attack against disabled users. In this logic, the number
+  of unique users is hardcoded to 10 but can or should be customized to fit especific
+  environments. Event 4768 is generated every time the Key Distribution Center issues
+  a Kerberos Ticket Granting Ticket (TGT). Failure code 0x12 means `clients credentials
+  have been revoked` (account disabled, expired or locked out).\
+
+  This detection will trigger on the domain controller against which the offending
+  host performs the password spraying attack.\
+
+  The analytics returned fields allow analysts to investigate the event further by
+  providing fields like source ip and attempted user accounts.'
+search: '`wineventlog_security` EventCode=4768 Account_Name!="*$" Result_Code=0x12
+  | bucket span=2m _time | stats dc(Account_Name) AS unique_accounts values(Account_Name)
```

Comment:
Misko, on my initial testing once I was working with the STDEV approach, I was using a larger bucket, I think 15 minutes. I found several false positives with that. I tested many different combinations on Dev RZ and uba-perf, the best results came with 2 minutes buckets.

On most of the scenarios I have encountered password spraying attacks, they are usually performed in a small period of time as attackers target a large number of users to increase the chances of finding valid credentials. Thats why I think it makes sense to measure what is "normal" in a small bucket of time.

Furthermore, we plan to configure this search so that it runs every hour. So, there will be 30 two-minute slots of time to identify observations and calculate the standard deviation. If we change this bucket to 30 minutes. there would only be 2 slots of time and this would affect the standard deviation calculation.